rholang/src/main/scala/coop/rchain/rholang/interpreter/util/Address.scala

```diff
@@ -1,3 +1,17 @@
 package coop.rchain.rholang.interpreter.util
 
-object Address {}
+import java.nio.charset.StandardCharsets
+
+import coop.rchain.crypto.hash.Blake2b256
+import coop.rchain.rholang.interpreter.util.codec.Base58
+
+object Address {
+  def fromPublicKey(prefix : String, pk : Array[Byte]) : String = {
```

Comment:
What is the purpose of the prefix?

Reply:
the prefix is a human-readable component of the address and its purpose is to be obvious what coin the address is referring to.

For example we can have REVAddress which would use the prefix "rev" or "testnet-rev" or maybe "rev:" ... that's something to decide.

The important things are that the prefix is ASCII so it's human readable, that the tools can do an initial validation of the address and that the checksum validates the prefix too so one could not copy a "revtest" address and use it to transfer "rev" to it.

Reply:
We should consider constraining the prefix length (max 16 chars?).

We also should also consider restricting it to alphanumeric characters only (so no `:` or `-` or similar) so that it's easier to validate the whole address string. If we allow anything more for the prefix than we allow for the suffix, there are two different alphabets to be checked against, and the prefix length is determined by subtracting the suffix length from total length, and the valid suffix lengths can (?) be many... I think you get my point that this starts to become complex if we allow more than base58 for prefix.

@cboscolo what are your thoughts? (re: prefix length and prefix alphabet)

Reply:
We did discuss this. But, as I ponder this thread, I think we should take a slightly more straight-forward approach that more closely follows the bitcoin approach.
To simplify decoding, I think the base58 step should only happen ONCE at the end of encoding, and the first step of decoding. (No concatenating base58 output.)
We should use a 4-byte prefix that consists of a 3-byte magic number and a 1-byte REV address version.
The 3-byte magic number should be chosen to result in an eye-pleasing base58 output.
I will update the Wiki

Reply:
The 3-byte magic number will be difficult to choose. 

See the conversions below:

hex                   Base58
================
028C7B      => rev
028C7B00 => 4nALw
028C7BF0 => 4nAR5
028C7BFF => 4nARL

For this reason I think the ASCII concatenation is better